Python/import.c

```diff
@@ -2249,6 +2249,9 @@ _imp_create_dynamic_impl(PyObject *module, PyObject *spec, PyObject *file)
 
     mod = _PyImport_FindExtensionObject(name, path);
     if (mod != NULL || PyErr_Occurred()) {
+        if (PyErr_Occurred()) {
```

Comment:
To avoid the overhead of calling `PyErr_Occurred()`, could you store the result in a variable and check that?